dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/DefaultProgramRuleEntityMapperService.java

```diff
@@ -109,7 +109,7 @@
 
     private final ImmutableMap<ProgramRuleVariableSourceType, Function<ProgramRuleVariable, RuleVariable>> VARIABLE_MAPPER = new ImmutableMap.Builder<ProgramRuleVariableSourceType, Function<ProgramRuleVariable, RuleVariable>>()
         .put( ProgramRuleVariableSourceType.CALCULATED_VALUE,
-            prv -> RuleVariableCalculatedValue.create( prv.getName(), "", RuleValueType.TEXT ) )
+            prv -> RuleVariableCalculatedValue.create( prv.getName(), prv.getUid(), toMappedValueType( prv ) ) )
```

Comment:
Just curious, since the PR title says add tests and this is non-test code. What behavior is it changing?

Reply:
what are these non-test changes doing?

Reply:
Passing the new ValueType attribute  (included in ProgramRuleVariable) to its associated object in the rule-engine library

Reply:
It just confused me because your PR title/description just says "add tests". Without more context (description, Jira ticket) its hard to say as a reviewer whether whats wanted is implemented.